documents/decorators.py

```diff
@@ -0,0 +1,19 @@
+from flask import jsonify, request
+import documents.state_machine as state_machine
+
+def auth(func):
+    def inner():
+        auth_header = request.headers.get('Authorization')
+        payload = state_machine.token_valid_check(auth_header)
+        return func(payload)
```

Comment:
what if the URL has named params, how are those passed to the lower function? like: /documents/:id ? I don't think this works